Common/Source/Draw/LKDrawWaypoints.cpp

```diff
@@ -694,8 +708,167 @@ void MapWindow::DrawWaypointPictoBg(LKSurface& Surface, const RECT& rc) {
     Surface.DrawBitmapCopy(x, y, cx, cy, hLKPictori, cxSrc, cySrc);
 }
 
+
+
+
+
+LKColor GetUTF8WaypointSymbol(TCHAR* pPict, const int Style)
+{
+
+LKColor Col = RGB_BLACK;
+if (pPict ==NULL) return Col;
+  switch(Style)
+  {
+    case STYLE_NORMAL:
+      _stprintf(pPict, _T("◎"));
+      Col = RGB_DARKGREEN;
+    break;
+
+    // These are not used here in fact
+    case STYLE_AIRFIELDGRASS:
+    case STYLE_OUTLANDING:
+    case STYLE_GLIDERSITE:
+    case STYLE_AIRFIELDSOLID:
+      _stprintf(pPict, _T("✈"));
+      Col = LKColor(131,111,255);
+    break;
+
+    case STYLE_MTPASS:
+      _stprintf(pPict, _T("◠"));
+      Col = RGB_ORANGE;
+    break;
+
+    case STYLE_MTTOP:
+      _stprintf(pPict, _T("▴"));
+      Col = RGB_BLACK;
+    break;
+
+    case STYLE_SENDER:
+      _stprintf(pPict, _T("⭻"));
+      Col = RGB_DARKGREY;
+    break;
+
+    case STYLE_VOR:
+      _stprintf(pPict, _T("⬡"));
+      Col = RGB_DARKGREY;
+    break;
+
+    case STYLE_NDB:
+      _stprintf(pPict, _T("✺"));
+      Col = RGB_DARKGREY;
+    break;
+
+    case STYLE_COOLTOWER:
+      _stprintf(pPict, _T("♜"));
+      Col = LKColor(82,82,82);
+    break;
+
+    case STYLE_DAM:
+      _stprintf(pPict, _T("⛊"));
+      Col = RGB_DARKBLUE;
+    break;
+
+    case STYLE_TUNNEL:
+      _stprintf(pPict, _T("⯊"));
+      Col = RGB_BLACK;
+    break;
+
+    case STYLE_BRIDGE:
+      _stprintf(pPict, _T("⋒"));
+      Col = RGB_DARKGREY;
+    break;
+
+    case STYLE_POWERPLANT:
+      _stprintf(pPict, _T("⚡"));
+      Col = LKColor(82,82,82);
+    break;
+
+    case STYLE_CASTLE:
+      _stprintf(pPict, _T("⛪"));
+      Col = LKColor(139,54,38);
+    break;
+
+    case STYLE_INTERSECTION:
+      _stprintf(pPict, _T("⛌"));
+      Col = RGB_BLACK;
+    break;
+
+    case STYLE_TRAFFIC:
+      _stprintf(pPict, _T("⛟"));
+      Col = RGB_BLACK;
+    break;
+    case STYLE_THERMAL:
+      _stprintf(pPict, _T("♨"));
+      Col = LKColor(210,105,30);
+    break;
+
+    case STYLE_MARKER:
+      _stprintf(pPict, _T("⚑"));
+      Col = LKColor(199,21,133);
+    break;
+
+    default:
+      _stprintf(pPict, _T("✈"));
+      Col = LKColor(199,21,133);
+    break;
+  } // switch estyle
+#ifdef KOBO
+  Col = LKColor(30,30,30);
+#endif
+return Col;
+}
+
```

Comment:
string literal containing non usascii characters are not crossplatform safe, neet bo be moved in language file